### ISSUE TITLE
sadc - high number of loop devices cause high cpu utilization

### ISSUE THREAD

**Issue by Pitterling** [sysstat/sysstat#183]
a high number of defined loop devices causes high CPU utilization by sadc.
This is even worse if you have several sar commands running in parallel.

This system was a SLES 11.4 XEN dom0 domain, which is using loop devices for communication with the domU domains. Therefore a high max_loop value has been configured

```console
# cat /sys/module/loop/parameters/max_loop 
6144 or 12228
```
The attached loop devices are only in a lower 2 digit range ... <40. 
```console
# losetup --all |wc -l
24 
```

But it seems that there sadc is iterating through all loop devices, which seems to be quite expansive.

**Comment by sysstat** [comment]
Please I need more details:
1) Which version of sadc are you using? (Enter ``sadc -V``)
2) How do you call sadc? Give all the options and parameters used.
3) Can you measure the CPU actually used by sadc? (Try to enter ``time sadc`` followed by the options and parameters you are using with sadc)
Is the amount of used CPU approximately the same if you run sadc (with its options) 2 or 3 times consecutively?
4) Also give me the exact output of the following commands: ``losetup --all`` ; ``df`` ; ``cat /etc/mtab``

Thanks.

**Comment by sysstat** [comment]
Sysstat 8.1.5 is *really* old (almost 10 years old) so I won't fix anything for this version.
Each call to sar actually ends up starting its data collector sadc which collects all the stats you need at the same time. I will try to do some code profiling with the latest sadc version and see where things can be improved.

**Comment by Pitterling** [comment]
I didn't expect a fix for this old version. Yes, unfortunately the Enterprise distributions are shipped with ancient sysstat version. Thanks for your support 👍

**Comment by sysstat** [comment]
FYI: Starting with sysstat version 11.7.2, sadc will collect only the activities you have selected on the command line with sar (unless data are saved into a file with option -o, in which case all the activities are read so that the datafile contains a full view of the system).
So if you enter `sar -n DEV -n EDEV 10 6` (in your example) then only statistics for network devices will be read.
`sar -n SOCK` will only read data for network sockets.
Etc.
This should greatly improve performance in your case.

**Comment by Pitterling** [comment]
👍 Thanks alot
